src/utils/path.py

```diff
@@ -1,14 +1,46 @@
 # Python Imports
+import os
+import logging
 from pathlib import Path
+from typing import List
 from result import Result, Err, Ok
 
 
-def prepare_path(folder_location: str, file_name: str) -> Result[Path, str]:
-    output_dir = Path(folder_location)
+logger = logging.getLogger(__name__)
+
+
+def prepare_path(file_location: str) -> Result[Path, str]:
+    file_location_path = Path(file_location)
+    dirname = get_parent_folders_from_file_path(file_location_path)
```

Comment:
`Path` to the rescue 😆
```suggestion
dirname = file_location_path.parent
```
Which given you wanna return `file_location_path`, means that instead of this assigment, you can just replace the line 20 for
```py
file_location_path.parent.mkdir(parents=True, exist_ok=True)
```
And kill 2 birds with one stone

Reply:
Yes indeed